include/linker/common-rom.ld

```diff
@@ -138,6 +138,15 @@
 	Z_ITERABLE_SECTION_ROM(dns_sd_rec, 4)
 #endif
 
+#if defined(CONFIG_PCIE)
```

Comment:
I made it for `CONFIG_PCIE` since that way we don't need #ifdefs in `drivers/pcie/host/pcie.c` (since the entire file itself is conditioned on `CONFIG_PCIE=y`). In the worst case non-X86 platforms supporting PCIe might just become slightly less broken since they will attempt allocation of IRQs for devices that do not have one pre-assigned. The only thing they won't have (until it's implemented) is the support through the `IRQ_CONNECT()` macro, which means that there could be collisions, however to my understanding that's primarily an efficiency issue (executing ISRs when not necessary).

Reply:
Strictly yes: there are devices out there with PCI buses that aren't x86.  None that Zephyr supports, but there's no point in putting in needless architecture dependencies.